web/packages/design/src/StepSlider/StepSlider.tsx

```diff
@@ -390,14 +405,14 @@ type Props<Flows> =
          */
         defaultStepIndex?: number;
       } & ExtraProps // Extra props that are passed to each step component. Each step of each flow needs to accept the same set of extra props.
-    : never;
+    : any;
```

Comment:
Aaah it's killing me, why does `never` not work here?

I've tried so many things from different SO threads to no avail. Adding `@types/react` and `@types/react-dom` to the design package, setting the `paths` in the tsconfig of design.

But the story of StepSlider still returns an error while the component itself seems to work just fine everywhere else.

Reply:
>But the story of StepSlider still returns an error while the component itself seems to work just fine everywhere else.

I think it's because other usages are not checked with `strict`. 

AFAIK `never` doesn't work because it's not allowed to have `props` of such type:
```tsx
const Child: React.FC<never> = () => {
  return <div />;
};

function Parent() {
  return <Child />;
}
```
In the strict mode I got two errors:
> TS2322: Type {} is not assignable to type never
TS2786: Child cannot be used as a JSX component.
Its type FC<never> is not a valid JSX element type.

In the non-strict mode only one:
>TS2322: Type {} is not assignable to type never
 
OTOH, I still don't understand why the conditional type goes to the `never` branch since it has the props specified statically (?)

Reply:
Oh I forgot that others don't use `strict`. This explains why those don't cause errors, but the errors we do get here is weird anyway.

```
web/packages/design/src/StepSlider/StepSlider.story.tsx:39:8 - error TS2786: 'StepSlider' cannot be used as a JSX component.
  Its type '<Flows>(props: Props<Flows>) => Element' is not a valid JSX element type.
    Type '<Flows>(props: Props<Flows>) => Element' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
      Types of parameters 'props' and 'props' are incompatible.
        Type 'any' is not assignable to type 'never'.

39       <StepSlider<typeof singleFlow>
```

Fortunately, from what I've seen, changing to `any` is not that big of a problem here. If the flow includes a step component that has extra props that are not passed to `StepSlider` itself, TS will still raise an error.

The only thing that `any` permits here is that individual step components are free to ignore extra props passed to other components. E.g., you can have one step that accepts only `StepComponentProps` and another that accepts `StepComponentProps & { foo: string }` and the type will allow this.